environment/src/main/java/com/sequenceiq/environment/environment/service/SdxService.java

```diff
@@ -0,0 +1,101 @@
+package com.sequenceiq.environment.environment.service;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BiFunction;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+import javax.ws.rs.WebApplicationException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import com.dyngr.Polling;
+import com.dyngr.core.AttemptMaker;
+import com.dyngr.exception.PollerException;
+import com.sequenceiq.cloudbreak.common.exception.WebApplicationExceptionMessageExtractor;
+import com.sequenceiq.environment.environment.poller.SdxPollerProvider;
+import com.sequenceiq.environment.exception.EnvironmentServiceException;
+import com.sequenceiq.sdx.api.endpoint.SdxEndpoint;
+import com.sequenceiq.sdx.api.model.SdxClusterResponse;
+
+@Service
+public class SdxService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SdxService.class);
+
+    @Value("${env.stop.polling.attempt:90}")
+    private Integer attempt;
+
+    @Value("${env.stop.polling.sleep.time:5}")
+    private Integer sleeptime;
+
+    private final SdxEndpoint sdxEndpoint;
+
+    private final SdxPollerProvider sdxPollerProvider;
+
+    private final WebApplicationExceptionMessageExtractor webApplicationExceptionMessageExtractor;
+
+    public SdxService(SdxEndpoint sdxEndpoint, SdxPollerProvider sdxPollerProvider,
+            WebApplicationExceptionMessageExtractor webApplicationExceptionMessageExtractor) {
+        this.sdxEndpoint = sdxEndpoint;
+        this.sdxPollerProvider = sdxPollerProvider;
+        this.webApplicationExceptionMessageExtractor = webApplicationExceptionMessageExtractor;
+    }
+
+    public void startAttachedDatalake(Long envId, String environmentName) {
+        executeSdxOperationAndStartPolling(envId, environmentName, sdxEndpoint::startByCrn, sdxPollerProvider::startSdxClustersPoller);
+
+    }
+
+    public void stopAttachedDatalakeClusters(Long envId, String environmentName) {
+        executeSdxOperationAndStartPolling(envId, environmentName, sdxEndpoint::stopByCrn, sdxPollerProvider::stopSdxClustersPoller);
+    }
+
+    private void executeSdxOperationAndStartPolling(Long envId, String environmentName, Consumer<String> sdxOperation,
+            BiFunction<Long, List<String>, AttemptMaker<Void>> attemptMakerFactory) {
+        try {
+            ArrayList<String> sdxCrns = getExecuteSdxOperationsAndGetCrns(environmentName, sdxOperation);
+            Polling.stopAfterAttempt(attempt)
+                    .stopIfException(true)
+                    .waitPeriodly(sleeptime, TimeUnit.SECONDS)
+                    .run(attemptMakerFactory.apply(envId, sdxCrns));
+        } catch (PollerException e) {
+            if (e.getCause() != null && e.getCause() instanceof WebApplicationException) {
```

Comment:
either rethrow this exception or leave a comment in the code if the condition is false